src/settings/lib/RuleEditor/Rules-hint.js

```diff
@@ -150,18 +182,61 @@ export default function rulesHint(Codemirror, props) {
     }
 
     if (result.length) {
+      sections.forEach(section => {
+        section.header = section.header ? formatMessage({ id: section.header }) : section.header;
+
+        if (section.selectedHintIndex === undefined) {
+          section.selectedHintIndex = -1;
+        }
+      });
+      sections[0].list = result;
```

Comment:
Please add a comment or move to method why it is needed to have `result` in the first section list.

Reply:
The comment was added. result contains the list of items properties that will be displayed in the first section. The other sections are empty by default and be filled by request in the separated helper.